### PR TITLE
Use learn from keyboard view

### PR DESCRIPTION
**Pull request by topisani**
Another minor QOL thing ive been missing - being able to use the LEARN button in keyboard view.
It works with the new kit keyboard view as well.

While implementing this, i found a few places in the code where it still assumed that kits could not have a keyboard view. This means for instance, that a synth can now be changed to a kit by pressing the KIT button in keyboard view (as it can when being in normal instrument view)

### CODE REVIEW

File: src/deluge/gui/ui/keyboard_screen.cpp [hunk 1]
```diff
@@ -404,12 +413,11 @@ void KeyboardScreen::selectEncoderAction(int8_t offset) {
 int KeyboardScreen::getNoteCodeFromCoords(int x, int y) {
 
 	Instrument* instrument = (Instrument*)currentSong->currentClip->output;
```
Comment (bobtwinkles) [inline src/deluge/gui/ui/keyboard_screen.cpp:415]: can set `instrument` from `clip` here I think.

File: src/deluge/gui/ui/keyboard_screen.cpp [hunk 2]
```diff
@@ -430,13 +438,18 @@ void KeyboardScreen::exitAuditionMode() {
...
-	for (int p = 0; p < MAX_NUM_KEYBOARD_PAD_PRESSES; p++) {
-		if (padPresses[p].x != 255) {
-			int noteCode = getNoteCodeFromCoords(padPresses[p].x, padPresses[p].y);
-			((MelodicInstrument*)currentSong->currentClip->output)->endAuditioningForNote(modelStack, noteCode);
-			if (switchOffOnThisEndToo) padPresses[p].x = 255;
+	if (currentSong->currentClip->output->type != INSTRUMENT_TYPE_KIT) {
+		for (int p = 0; p < MAX_NUM_KEYBOARD_PAD_PRESSES; p++) {
+			if (padPresses[p].x != 255) {
+				int noteCode = getNoteCodeFromCoords(padPresses[p].x, padPresses[p].y);
+				((MelodicInstrument*)currentSong->currentClip->output)->endAuditioningForNote(modelStack, noteCode);
```
Comment (bobtwinkles) [inline src/deluge/gui/ui/keyboard_screen.cpp:445]: `currentSong->currentClip` => `getCurrentClip()`

File: src/deluge/gui/ui/keyboard_screen.cpp [hunk 3]
```diff
@@ -430,13 +438,18 @@ void KeyboardScreen::exitAuditionMode() {
...
+				if (switchOffOnThisEndToo) padPresses[p].x = 255;
+			}
 		}
 	}
+	else {
+		// TODO: Implement for kits
```
Comment (bobtwinkles) [inline src/deluge/gui/ui/keyboard_screen.cpp:451]: Maybe flash a `SOON` here?

Comment (topisani) [reply src/deluge/gui/ui/keyboard_screen.cpp:451]: I dont think so, as this function is called mostly from within other routines, so this message would pop up at random-ish times for the user

**Comment by ok-reza** [comment]
What steps can be made to get this merged?

**Comment by PaulFreund** [comment]
Hey, a quick heads up :) After two weeks without any activity from the author of a PR we tag it with the "stale" label and close it after another week of inactivity as per our [Contributor guidelins](https://github.com/SynthstromAudible/DelugeFirmware/blob/community/docs/CONTRIBUTING.md?plain=1#L50). This is not to create pressure or discourage anyone but to keep the list of Pull requests maintainable.

We are very happy about your contribution and nothing will be lost. You can pick up the work at any time by reopening the PR or creating a new one.

**Pull request closed without merging**